Sources/Kafka/KafkaConsumer.swift

```diff
@@ -331,6 +324,13 @@ public final class KafkaConsumer: Sendable, Service {
     }
 
     private func _run() async throws {
+        switch self.configuration.consumptionStrategy._internal {
+        case .partition(topic: let topic, partition: let partition, offset: let offset):
+            try self.assign(topic: topic, partition: partition, offset: offset)
+        case .group(groupID: _, topics: let topics):
+            try self.subscribe(topics: topics)
+        }
```

Comment:
We should probably go through the state machine here

Reply:
`func assign` and `func subscribe` already go through the state machine by invoking `stateMachine.setupConnection`

I could however hoist the state check here (so `state check -> check consumption strategy -> assign / subscribe` instead of `check consumption strategy -> state check -> assign / subscribe`

Reply:
Yes, it only works when in state `.initializing` 👍